common/src/main/java/com/microsoft/identity/common/internal/telemetry/AndroidTelemetryContext.java

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.internal.telemetry;
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.os.Build;
+
+import androidx.core.content.pm.PackageInfoCompat;
+
+import com.microsoft.identity.common.internal.cache.SharedPreferencesFileManager;
+import com.microsoft.identity.common.java.interfaces.IKeyPairStorage;
+import com.microsoft.identity.common.java.telemetry.AbstractTelemetryContext;
+import com.microsoft.identity.common.java.telemetry.TelemetryEventStrings;
+import com.microsoft.identity.common.logging.Logger;
+
+import lombok.NonNull;
+
+/**
+ * TelemetryContext for Android.
+ * Containing Android Metadata. It also persists data in SharedPreferences.
+ * */
+public class AndroidTelemetryContext extends AbstractTelemetryContext {
+
+    private static final String TAG = AndroidTelemetryContext.class.getName();
+    private static final String SHARED_PREFS_NAME = "com.microsoft.common.telemetry-properties";
+
+    public AndroidTelemetryContext(@NonNull final Context context) {
+        super(
+                new IKeyPairStorage() {
+                    final SharedPreferencesFileManager mSharedPrefs =
+                            SharedPreferencesFileManager.getSharedPreferences(context, SHARED_PREFS_NAME, -1, null);
+
+                    @Override
+                    public String get(@NonNull String key) {
+                        return mSharedPrefs.getString(key);
+                    }
+
+                    @Override
+                    public void put(@lombok.NonNull String key, String value) {
+                        mSharedPrefs.putString(key, value);
+                    }
+                });
```

Comment:
Can we put this implementation in its own class?

Reply:
I'd say that this adapter is where it belongs - for now. Eventually - we want to have this set in the beginning of the flow.

Reply:
yeah I was basically referring to whether we should create this as this inline-class or if we can just put this in a separate java class and then over here you can just create its instance....so over here your code would reduce from 14 lines to 1 line